drivers/i2c/i2c_andes_atciic100.c

```diff
@@ -0,0 +1,789 @@
+/*
+ * Copyright (c) 2022 Andes Technology Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file I2C driver for AndesTech atciic100 IP
+ */
+
+#include <string.h>
+#include <zephyr/drivers/i2c.h>
+#include "i2c_andes_atciic100.h"
+
+#define DT_DRV_COMPAT andestech_atciic100
+
+typedef void (*atciic100_dt_init_func_t)(void);
+
+struct i2c_atciic100_device_config {
+	uint32_t			i2c_base_addr;
+	uint32_t			i2c_irq_num;
+	atciic100_dt_init_func_t	dt_init_fn;
+};
+
+static int i2c_atciic100_controller_send(const struct device *dev,
+	uint16_t addr, const uint8_t *data, uint32_t num, uint8_t flags);
+static int i2c_atciic100_controller_receive(const struct device *dev,
+	uint16_t addr, uint8_t *data, uint32_t num, uint8_t flags);
+static void i2c_controller_fifo_write(const struct device *dev,
+	uint8_t is_init);
+static void i2c_controller_fifo_read(const struct device *dev);
+static int i2c_atciic100_init(const struct device *dev);
+
+#if defined(CONFIG_I2C_TARGET)
+static void i2c_atciic100_target_send(const struct device *dev,
+		const uint8_t *data);
+static void i2c_atciic100_target_receive(const struct device *dev,
+		uint8_t *data);
+#endif
+
+static void i2c_atciic100_default_control(const struct device *dev)
+{
+	struct i2c_atciic100_dev_data_t *dev_data = DEV_I2C_DATA(dev);
+	uint32_t reg = 0;
+
+	k_sem_init(&dev_data->i2c_busy_sem, 0, 1);
+	k_sem_give(&dev_data->i2c_busy_sem);
```

Comment:
I just wonder why the `k_sem_init` followed by a `k_sem_give` instead of initialising it to `1`?

Reply:
We have no particular reason for that, but it is indeed better to initialize it to 1. Thanks for the suggestion.